### PR TITLE
virtio_fs: add user to the group when it exists

### PR DESCRIPTION
**Pull request by xiagao**
Sometimes when add a new user, it failed when the group
    existed. In this situation, add the new user to the group.

ID: 2224

### CODE REVIEW

**Comment by xiagao** [comment]
Hi @hellohellenmao , could you help review this one? Thanks.

**Comment by leidwang** [comment]
Hi @hellohellenmao 
Would you please help review this PR when you are free?Thanks.

**Comment by hellohellenmao** [comment]
@xiagao Sorry I missed your comment. I will do it ASAP. cc @leidwang

**Review by hellohellenmao — CHANGES_REQUESTED** [review]

File: qemu/tests/virtio_fs_group_permission_access.py [hunk 1]
```diff
@@ -61,6 +61,9 @@ def run(test, params, env):
                     else:
                         test.fail("Unknown error when deleting the "
                                   "user: %s" % o)
+            if process.system("grep %s /etc/group" % _username, shell=True,
+                              ignore_status=True) == 0:
+                process.run(params["del_group_cmd"] % _username)
```
[the remark this answers is not in the document]
Comment (xiagao) [reply qemu/tests/virtio_fs_group_permission_access.py]: I prefer to add the user to the known group.
 Updated. Thx.

**Comment by hellohellenmao** [comment]
LGTM.

**Comment by xiagao** [comment]
(1/2) Host_RHEL.m9.u5.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2022.x86_64.io-github-autotest-qemu.virtio_fs_group_permission_access.q35: STARTED
 (1/2) Host_RHEL.m9.u5.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2022.x86_64.io-github-autotest-qemu.virtio_fs_group_permission_access.q35: PASS (140.00 s)
 (2/2) Host_RHEL.m9.u5.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2022.x86_64.io-github-autotest-qemu.virtio_fs_group_permission_access.with_nfs_source.q35: STARTED
 (2/2) Host_RHEL.m9.u5.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.Win2022.x86_64.io-github-autotest-qemu.virtio_fs_group_permission_access.with_nfs_source.q35: PASS (168.19 s)

**Comment by xiagao** [comment]
@YongxueHong Could you please help review? Thx.

File: qemu/tests/virtio_fs_group_permission_access.py [hunk 2]
```diff
@@ -61,7 +62,10 @@ def run(test, params, env):
                     else:
                         test.fail("Unknown error when deleting the "
                                   "user: %s" % o)
-            process.run(add_user_cmd % _username)
+            if process.system("grep %s /etc/group" % _username, shell=True,
+                              ignore_status=True) == 0:
+                add_cmd = "useradd -g %s %s" % (_username, _username)
+            process.run(add_cmd)
```
Comment (YongxueHong) [inline qemu/tests/virtio_fs_group_permission_access.py:68]: Hi @xiagao 
I found no steps for deleting the `usergroup` in this file, unlike the following file.

Comment (xiagao) [reply qemu/tests/virtio_fs_group_permission_access.py:68]: Generally speaking, the group is supposed to be deleted when delete user with cmd "    del_user_cmd = "userdel -r -f %s"
", but sometimes the group existed while the user was not there, therefore adding L65 to L67 to adapt such situation.

**Review by YongxueHong — APPROVED** [review]
LGTM.

**Pull request merged**